src/main/java/org/folio/circulation/domain/AddressTypeRepository.java

```diff
@@ -28,4 +33,29 @@ public AddressTypeRepository(Clients clients) {
       .whenNotFound(succeeded(null))
       .fetch(id);
   }
+
+  public CompletableFuture<Result<MultipleRecords<AddressType>>> getAddressTypesByIds(Collection<String> ids) {
+    return new MultipleRecordFetcher<>(addressTypesStorageClient, "addressTypes", AddressType::new)
+      .findByIds(ids);
+  }
+
+  public CompletableFuture<Result<MultipleRecords<Request>>> findAddressTypesForRequests(
+    MultipleRecords<Request> requests) {
+
+    Set<String> addressTypeIds = new HashSet<>(requests.toKeys(Request::getDeliveryAddressTypeId));
```

Comment:
Why is it important that this is a set, is that only to remove duplicate values?

Reply:
Yes.

Reply:
Please could you investigate changing the `toKeys` method to return a set instead, meaning this conversion can be avoided.

Reply:
done